### DISCUSSION TITLE
Allowing objectToFormData to be forced

### DISCUSSION THREAD

**Discussion by nickknissen** [inertiajs/inertia#405] [Ideas]
For dotnet projects you have to specify the data binding type (json/form data). 
Which makes it inconvenient to use a form where a a file is optional. Because you have to have multiple method for the same action.

Ex. multipart/form-data
```csharp
[HttpPut("{id}")]
[Consumes("multipart/form-data")]
public async Task<IActionResult> UpdateFormData([FromForm] OrganizationViewModel viewModel, Guid id)
{
...
```
Ex. json
```csharp
[HttpPut("{id}")]
[Consumes("application/json")]
public async Task<IActionResult> UpdateJson([FromBody] OrganizationViewModel viewModel, Guid id)
{
...
```

Allowing objectToFormData to be forced via options argument would help a lot with the boilerplate.

The change in inertia.js:
```js
visit(url, {
    ....
    errorBag = null,
    forceFormData = false,
    onCancelToken = () => ({}),
    ...
  } = {}) {
    method = method.toLowerCase();
    [url, data] = mergeDataIntoQueryString(method, hrefToUrl(url), data)

    const visitHasFiles = hasFiles(data)
    if (forceFormData || method !== 'get' && visitHasFiles) {
      data = objectToFormData(data)
    }
```

Example usage:
```js
this.form.post(this.route("organizations.store"), { forceFormData: true });
```

**Comment by claudiodekker** [comment]
@reinink Perhaps it would be a good idea to move the `transform` method out of the Form, and into the underlying visit?
That way, we don't have to introduce explicit support for specific edge cases like this while still solving the problem.

**Comment by reinink** [comment]
One way to solve this is to just manually do the `FormData` conversion yourself. That way it will always be sent as a `multipart/form-data` request, even if files are not present. That, obviously, doesn't help you take advantage of this feature of Inertia though.

**Comment by reinink** [comment]
I just put together a PR for this: https://github.com/inertiajs/inertia/pull/407
